### PR TITLE
set use_single_app_instance config through env var

### PR DESCRIPTION
**Pull request by nimakaviani**
Signed-off-by: Nima Kaviani <nkavian@us.ibm.com>

### CODE REVIEW

**Comment by sjolicoeur** [comment]
Please *do not* merge immediately, we just wanted to expose what we were doing, we still need to figure out how to solve for credhub

File: bosh-deploy-with-created-release/task.yml [hunk 1]
```diff
@@ -16,6 +16,7 @@ inputs:
 # - BOSH release source repo
 # - A dev release will be created from this repo and used in the deployment
 - name: cf-deployment-concourse-tasks  # - This repo
+- name: dev-release
```
Comment (sjolicoeur) [inline bosh-deploy-with-created-release/task.yml]: we needed an extra input here to be able to pass our release folder

File: shared-functions [hunk 2]
```diff
@@ -92,15 +92,30 @@ function bosh_interpolate() {
     arguments="${arguments} -l vars-files/${vf}"
   done
 
-  if [ -n "${release_name}" ]; then
+  if [ -z "${RELEASE_TARBALL}" ]; then
```
Comment (sjolicoeur) [inline shared-functions]: we check if the pipeline has set a `RELEASE_TARBALL` env var and then go to a flow that we used to have. If there is a better way, suggestions are welcomed.

File: shared-functions [hunk 3]
```diff
@@ -92,15 +92,30 @@ function bosh_interpolate() {
...
+    path: /releases/name=${release_name}
+    value:
+      name: ${release_name}
+      version: create
+      url: file://${root_dir}/release
+EOF
+      arguments="${arguments} -o create-provided-release.yml"
+    fi
+  else
+    RELEASE_TARBALL=$(readlink -f -- ${RELEASE_TARBALL})
```
Comment (sjolicoeur) [inline shared-functions]: we were using `realpath` but it seems it's not there by default and is installed as part of the core-utils on ubuntu. I think the default docker image may not have it, so `readlink` works. 
it allows us to complete paths that are like `../diego-release/*.tgz` to `/tmp/workspace/diego/diego-release.2.2.2.tgz` for example.

File: shared-functions [hunk 4]
```diff
@@ -136,6 +151,9 @@ write_uptimer_deploy_config() {
   if [ -z "${AVAILABLE_PORT}" ]; then
     AVAILABLE_PORT="-1"
   fi
+  if [ -z "${USE_SINGLE_APP_INSTANCE}" ]; then
```
Comment (sjolicoeur) [inline shared-functions]: this is for the uptimer option we so desired!

**Comment by heyjcollins** [comment]
@sjolicoeur - since you want us to hold on integrating this, do you have tracker story that we can block this on?

**Comment by sjolicoeur** [comment]
@heyjcollins  here's the link to our story: https://www.pivotaltracker.com/story/show/159444743

**Comment by sjolicoeur** [comment]
And we are good to proceed with the PR

File: update-integration-configs/task [hunk 5]
```diff
@@ -5,8 +5,8 @@ set -o pipefail
 source cf-deployment-concourse-tasks/shared-functions
 
 function check_fast_fails() {
-  if [ ! -f "integration-configs/$CATS_INTEGRATION_CONFIG_FILE" -a ! -f "integration-configs/$RATS_INTEGRATION_CONFIG_FILE" ]; then
-    echo "Neither cats_integration_config file nor rats_integration_config file is present...Exiting"
+  if [ ! -f "integration-configs/$CATS_INTEGRATION_CONFIG_FILE" -a ! -f "integration-configs/$RATS_INTEGRATION_CONFIG_FILE" -a ! -f "integration-configs/$WATS_INTEGRATION_CONFIG_FILE" ]; then
```
Comment (sjolicoeur) [inline update-integration-configs/task]: @ishustava  this was the change I mentioned

**Comment by nimakaviani** [comment]
closing in favor of the upcoming PRs.

**Pull request closed without merging**